File: Daily Test/Java/Sort Subarrays By First Integer 10.10.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] arr = new int[N];
        for(int i=0;i<N;i++) arr[i]=sc.nextInt();
        int K = sc.nextInt();

        List<List<Integer>> res = new ArrayList<>();
        for(int i=0;i<N;i+=K)
            res.add(Arrays.asList(Arrays.copyOfRange(arr, i, Math.min(i+K,N))));
        res.sort(Comparator.comparingInt(a -> a.get(0)));
        for(List<Integer> block: res)
            for(int x:block)
                System.out.print(x + " ");
    }
}

File: Daily Test/C++/Sort Subarrays By First Integer 10.10.2025.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

bool cmp(const vector<int>& a, const vector<int>& b) {
    return a[0] < b[0];
}

int main() {
    int N, K;
    cin >> N;
    vector<int> arr(N);
    for(int i = 0; i < N; ++i)
        cin >> arr[i];
    cin >> K;
    vector<vector<int>> res;
    for(int i = 0; i < N; i += K)
        res.push_back(vector<int>(arr.begin() + i, arr.begin() + min(i+K,N)));
    sort(res.begin(), res.end(), cmp);
    for(const auto& block : res)
        for(int x : block)
            cout << x << ' ';
    return 0;
}